trace/header.go

```diff
@@ -1,34 +1,60 @@
 package trace
 
 import (
+	"log"
 	"net/http"
-
-	"github.com/twinj/uuid"
 )
 
 const (
-	HeaderId = "x-request-id"
-
-	HeaderParentId = "x-parent-id"
+	// HeaderSpanID is the header name in requests for the SpanID.
+	HeaderSpanID = "Trace-Span"
+	// HeaderTraceID is the header name in requests for the TraceID.
+	HeaderTraceID = "Trace-Trace"
 )
 
-func GetId(r *http.Request) string {
-	return r.Header.Get(HeaderId)
+// Generates a new SpanID from a request. If there is no span information in the request, a root SpanID will be generated.
+func NewSpanIDFromRequest(req *http.Request) SpanID {
+	parentSpanID := ParseSpanID(req)
+	if parentSpanID.Empty() {
+		return NewRootSpanID()
+	}
+	return NewSpanID(parentSpanID)
+}
+
+// Returns the SpanID contained in the request.
+func ParseSpanID(req *http.Request) SpanID {
+	spanIDStr := req.Header.Get(HeaderSpanID)
+	traceIDStr := req.Header.Get(HeaderTraceID)
+	spanID, err := ParseID(spanIDStr)
+	if err != nil {
+		log.Println("Failed to parse spanID string value, using 0.")
```

Comment:
I documented this behaviour. The good solution should be to differentiate between missing header and failed parsing. I will change this behaviour later.